:sparkles: Improve logging in core components

ff137 opened the pull request:
List of changes:
- Expands logging within core and config components: Conductor, PluginRegistry, and DefaultContextBuilder
- Replaces (almost) all print statements with info logs
- General improvement of some code for readability / maintainability (mainly in the PluginRegistry)
- Includes unit test coverage 

Comparison of start up logs before and after to be included.

Partially resolves #3202

Note:
- reviewing the logs, I can see a lot of inefficiency in the number of times that `ClassLoader.load_module` is called. We see the log "Module ... is already loaded" 500 times during startup. And because I've added more verbose debug logging, there are 5 logs between trying to load it and recognising that it's already loaded. So, currently my log changes will introduce 2500 unnecessary log lines. As a solution, I'll implement caching to the load_module function, so it doesn't execute unnecessarily.
- Additionally, the logging now reveals that upon every /status check, the AskarStorage module is attempted to be re-loaded ... representing some inefficiency. Will see if I can resolve that.

ff137 commented:
Not sure why the following Scenario Integration Tests are failing:
```sh
FAILED examples/anoncreds_issuance_and_revocation::anoncreds_issuance_and_revocation
FAILED examples/presenting_revoked_credential::presenting_revoked_credential
FAILED examples/self_attested::self_attested
```

ff137 commented:
Looks like BDD Interop tests also silently fail: ("silently" because the Github action shows a green tick)
```sh
Failing scenarios:
  features/0036-issue-credential.feature:9  Issue a credential with the Holder beginning with a proposal
  features/0036-issue-credential.feature:23  Issue a credential with the Holder beginning with a proposal with DID Exchange Connection
  features/0036-issue-credential.feature:53  Issue a credential with the Issuer beginning with an offer
  features/0037-present-proof.feature:19  Present Proof where the prover does not propose a presentation of the proof and is acknowledged -- @1.1 
  features/0037-present-proof.feature:20  Present Proof where the prover does not propose a presentation of the proof and is acknowledged -- @1.2 
  features/0037-present-proof.feature:37  Present Proof where the prover does not propose a presentation of the proof and is acknowledged with a DID Exchange Connection -- @1.1 
  features/0037-present-proof.feature:38  Present Proof where the prover does not propose a presentation of the proof and is acknowledged with a DID Exchange Connection -- @1.2 
  features/0037-present-proof.feature:55  Present Proof of specific types and proof is acknowledged with a Drivers License credential type -- @1.1 
  features/0037-present-proof.feature:56  Present Proof of specific types and proof is acknowledged with a Drivers License credential type -- @1.2 
  features/0037-present-proof.feature:74  Present Proof of specific types and proof is acknowledged with a Biological Indicators credential type -- @1.1 
  features/0037-present-proof.feature:91  Present Proof of specific types and proof is acknowledged with multiple credential types -- @1.1 
  features/0037-present-proof.feature:109  Present Proof where the prover does not propose a presentation of the proof and is acknowledged -- @1.1 
  features/0037-present-proof.feature:110  Present Proof where the prover does not propose a presentation of the proof and is acknowledged -- @1.2 
  features/0037-present-proof.feature:129  Present Proof where the prover has proposed the presentation of proof in response to a presentation request and is acknowledged -- @1.1 
  features/0037-present-proof.feature:130  Present Proof where the prover has proposed the presentation of proof in response to a presentation request and is acknowledged -- @1.2 
  features/0037-present-proof.feature:150  Present Proof where the prover has proposed the presentation of proof from a different credential in response to a presentation request and is acknowledged -- @1.1 
  features/0037-present-proof.feature:151  Present Proof where the prover has proposed the presentation of proof from a different credential in response to a presentation request and is acknowledged -- @1.2 
  features/0183-revocation.feature:79  Credential revoked and replaced with a new updated credential, holder proves claims with the updated credential with timesstamp -- @1.1 
  features/0183-revocation.feature:380  Issuer revokes a credential and sends a v1 revocation notification -- @1.1 
  features/0434-out-of-band.feature:9  Issue a v1 indy credential using connectionless out of band invitation
  features/0434-out-of-band.feature:50  Present a v1 indy proof using connectionless out of band invitation

7 features passed, 4 failed, 4 skipped
36 scenarios passed, 21 failed, 103 skipped
348 steps passed, 21 failed, 1014 skipped, 0 undefined
Took 13m2.976s
```

ff137 commented:
Yeah, those test failures are a real head-scratcher for me... Doesn't seem to occur in other PRs. And I can't see what in this contribution will cause the problem.

Maybe there's a way to print the ACA-Py agent logs as part of those Test workflows, see if that reveals anything.

Only solution I can think of is to rebase and drop / re-add commits until I can identify what causes the problem 😄

dbluhm commented:
Just remembered Jamie is out until Wednesday (I was hoping to just throw this his way lol); I'll take a deeper look at this. It should be possible to get some more logs on failures in the scenario tests, I think.

jamshale commented:
I'm not back until Friday but quickly looked at the logs. 2 of them are failing when trying to receive the revocation webhook notification. Perhaps the log level affected the library checking for this event.

One of the issues with using an external library for this is that we may need to get the library changed for changes like this.

For the self attest test, I can't really think of a reason off the top of my head. Running an individual test with higher logging could help find the issue.

dbluhm commented:
An unfortunate side effect of this change is that pyright/pylance does not detect `logger.trace` as a valid method

dbluhm commented on acapy_agent/core/plugin_registry.py:
```diff
@@ -216,134 +253,183 @@ async def load_protocol_version(
                 for plugin_version in definition.versions:
+                    version_path = f"{plugin_name}.{plugin_version['path']}.routes"
                     try:
-                        mod = ClassLoader.load_module(
-                            f"{plugin.__name__}.{plugin_version['path']}.routes"
-                        )
+                        LOGGER.trace("Loading routes from: %s", version_path)
+                        mod = ClassLoader.load_module(version_path)
```
We're overwriting `mod` with each version. Only the last mod's routes are actually registered.

ff137 replied:
Aah, that's what I get for micro-optimizing! I moved the `await mod.register(app)` step out of the for loop without noticing ... because it seemed  duplicated in both if and else blocks... Thanks for picking up on that. 

These methods also get flagged by sonarcloud for their cognitive complexity, due to all the nesting. So I want to refactor that for readability too :+1:

dbluhm commented on acapy_agent/core/plugin_registry.py:
```diff
@@ -216,134 +253,183 @@ async def load_protocol_version(
                 for plugin_version in definition.versions:
+                    version_path = f"{plugin_name}.{plugin_version['path']}.routes"
                     try:
-                        mod = ClassLoader.load_module(
-                            f"{plugin.__name__}.{plugin_version['path']}.routes"
-                        )
+                        LOGGER.trace("Loading events from: %s", version_path)
+                        mod = ClassLoader.load_module(version_path)
```
Similar problem; we're overwriting `mod` with each version. Only the last mod's events are actually registered.

dbluhm reviewed (changes requested):
Spotted the issue. See comments.

dbluhm commented:
While testing, I couldn't get the trace logs to print from the plugin registry. I tried setting the log level to "trace" (in the docker-compose for the anoncreds scenario test) and only saw debug and higher. Did I miss something?

ff137 commented:
> An unfortunate side effect of this change is that pyright/pylance does not detect `logger.trace` as a valid method. This results in a pretty significant amount of noise in editor, especially in modules with particularly heavy use of trace logs.

Hmm, that sucks... 

So, my initial idea was to try swap out the default python logger for [loguru](https://github.com/Delgan/loguru), which is much better and offers plenty of enhancements over the python logger. Loguru includes trace levels, and built in solutions for serialised logging and timed-file-logging config, etc, which was manually implemented here.

What that would look like is defining a centralised loguru "get_logger" method somewhere, with all the acapy-config, and importing that, instead of using `logging.getLogger`.

Some other nice features are colorized logging, and being able to bind context with logs. (e.g. `bound_logger = logger.bind(thread_id=thread_id)` and then you can implicitly include metadata in all logs when using bound_logger instead.)

I think that's still on the cards, because it really is more elegant to work with. But it does represent a big change to a core component (logging). And it'll take some effort to ensure everything remains identical. The potential bulk refactoring can cause some headaches for other maintainers with merge conflicts. And then there's effort in getting everyone on board with the change. 

So, at first I wanted to scrap the loguru idea, and just go for this trace addition... Buut, what do you think? Any potential appetite for switching to loguru? It'll take some more time, but I can take a swing at it.

ff137 commented:
> While testing, I couldn't get the trace logs to print from the plugin registry. I tried setting the log level to "trace" (in the docker-compose for the anoncreds scenario test) and only saw debug and higher. Did I miss something?

Hmm, it worked on my local with `ACAPY_LOG_LEVEL: trace`... 

Example for loading `protocols.endorse_transaction`
```
2024-11-07 19:55:39,251 acapy_agent.core.plugin_registry TRACE Processing events for plugin: acapy_agent.protocols.endorse_transaction
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Attempting to load module: definition with package: acapy_agent.protocols.endorse_transaction
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Preloading parent package: acapy_agent.protocols.endorse_transaction
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Attempting to load module: acapy_agent.protocols.endorse_transaction with package: None
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Resolved full module path: acapy_agent.protocols.endorse_transaction
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Module acapy_agent.protocols.endorse_transaction is already loaded
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Adjusted mod_path for relative import: .definition
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Resolved full module path: acapy_agent.protocols.endorse_transaction.definition
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Module acapy_agent.protocols.endorse_transaction.definition is already loaded
2024-11-07 19:55:39,251 acapy_agent.core.plugin_registry TRACE Loading versioned events for: acapy_agent.protocols.endorse_transaction
2024-11-07 19:55:39,251 acapy_agent.core.plugin_registry TRACE Loading events from: acapy_agent.protocols.endorse_transaction.v1_0.routes
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Attempting to load module: acapy_agent.protocols.endorse_transaction.v1_0.routes with package: None
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Resolved full module path: acapy_agent.protocols.endorse_transaction.v1_0.routes
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Parent module path: acapy_agent.protocols.endorse_transaction.v1_0, Module name: routes
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Attempting to load module: acapy_agent.protocols.endorse_transaction.v1_0 with package: None
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Resolved full module path: acapy_agent.protocols.endorse_transaction.v1_0
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Module acapy_agent.protocols.endorse_transaction.v1_0 is already loaded
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Adjusted mod_path after loading parent: .routes
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Finding spec for module: .routes with package: acapy_agent.protocols.endorse_transaction.v1_0
2024-11-07 19:55:39,251 acapy_agent.utils.classloader TRACE Importing module: .routes with package: acapy_agent.protocols.endorse_transaction.v1_0
2024-11-07 19:55:39,255 acapy_agent.core.plugin_registry TRACE Registering events from: acapy_agent.protocols.endorse_transaction.v1_0.routes
```

You can see that's why I didn't want it at debug level, because it'll clutter the existing, expected logs for debug.

dbluhm commented:
> So, at first I wanted to scrap the loguru idea, and just go for this trace addition... Buut, what do you think? Any potential appetite for switching to loguru? It'll take some more time, but I can take a swing at it.

I'm not opposed to the idea in principle but definitely agree that it could be a bit of a messy change to integrate with ongoing work streams.

How would this affect plugins? Would plugins also have to use loguru or is there a way to get loguru to integrate with Python's builtin logging?

Are there any performance implications to loguru over builtin logging module?

I think I'm also still a bit on the fence about the idea of a trace log level to begin with. Obviously it's not unheard of to have a level below debug if other loggers like loguru support it. Several of our rust based dependencies like Askar and Indy-VDR also support a trace log level. In the case of those dependencies, though, the inspect-ability of those dependencies is really low because they're compiled libraries. Python, on the other hand, being interpreted, it's pretty trivial to step into a debugger and see exactly what is going on when necessary. I'm a bit skeptical of the value of having the intense kind of detail being emitted to the trace level in these changes.

I think we can definitely up our logging game in ACA-Py but the kind of change I think I'm more interested in is probably more effective/judicious use of the INFO level. 90% of the time when I'm scanning through logs, I'm looking for what messages were sent and received or what Admin API requests came in and their responses. In other words, I usually want to filter through less information but I have to use debug level right now or I don't get what I'm after.

ff137 commented:
@dbluhm That makes sense. I think the trace level can be omitted... and instead, just add some sensible debug logs, and start the initiative to move more of the useful logs to info level.

The primary motivator here was to debug an issue with an agent not starting up at all. It was from testing out a fix for the RuntimeWarning from the IndyVdrLedgerPool (#3328), and there were no debug logs or errors to indicate what was causing the agent to hang.

Then I realised there is actually quite a lot of steps in the startup logic with no logging whatsoever. So, I just went at it, adding some sensible logging steps throughout. But, that made debug logging too verbose ... and figured, that is what trace is for.

Fortunately, I've already achieved the goal of isolating what was causing the issue. So, I'd say this particular way of implementing trace logging is optional. And, Loguru should be the preferred way to implement trace functionality.

I do think Loguru is worth it. The performance improvement is probably negligible ([until they implement it in C](https://github.com/Delgan/loguru?tab=readme-ov-file#10x-faster-than-built-in-logging)), and for plugins it shouldn't be any issue.

I'm imagining we can define a central, configured logging instance, similar to how we've done [here](https://github.com/didx-xyz/aries-cloudapi-python/blob/master/shared/log_config.py) - and then all cases of
```py
import logging
LOGGER = logging.getLogger(__name__)
```
can just be replace-all'd with
```py
from acapy_agent.config.logging import get_logger
LOGGER = get_logger(__name__)
```
Then everything else remains the same.

Using standard python logging would still be possible - it just won't benefit from the loguru features. So it would just require letting all the devs know, hey, use get_logger instead.

I think it's one of those "nice-to-haves" that once it's in place, it's a godsend, and makes developing so much easier -- lazy logging, context binding, colorized logging, and much easier configuration. There's just the effort in proving that it's a seamless shift, and that all existing logging configuration can be ported to the new way of doing things. Difficult to forecast how much effort that'll be. But I can try chip away at it, sometime over the next several weeks.

So, over the next while I'll try clean up this PR to just contribute some reasonable addition to the startup logs, and if it proves seamless enough, loguru can come later :-)

ff137 commented:
I'll wrap this one up sometime soon. I think these are worthwhile changes, which we've pulled into our agent. It helps identify some potential improvements, which, unresolved leads to some noisy debug logs. Will see if I can improve on that

ff137 commented:
Startup logs before (debug level):
```
2025-04-09 11:11:54,867 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.did_rotate
2025-04-09 11:11:54,868 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.present_proof
2025-04-09 11:11:54,868 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.routing
2025-04-09 11:11:54,868 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.didexchange
2025-04-09 11:11:54,868 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.introduction
2025-04-09 11:11:54,868 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.coordinate_mediation
2025-04-09 11:11:54,869 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.revocation_notification
2025-04-09 11:11:54,869 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.connections
2025-04-09 11:11:54,869 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.problem_report
2025-04-09 11:11:54,870 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.notification
2025-04-09 11:11:54,870 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.trustping
2025-04-09 11:11:54,870 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.issue_credential
2025-04-09 11:11:54,871 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.actionmenu
2025-04-09 11:11:54,871 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.out_of_band
2025-04-09 11:11:54,871 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.endorse_transaction
2025-04-09 11:11:54,871 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.basicmessage
2025-04-09 11:11:54,872 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.protocols.discovery
2025-04-09 11:11:54,872 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.holder
2025-04-09 11:11:54,884 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.ledger
2025-04-09 11:11:54,902 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.messaging.jsonld
2025-04-09 11:11:54,905 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.resolver
2025-04-09 11:11:54,905 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.settings
2025-04-09 11:11:54,912 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.vc
2025-04-09 11:11:54,912 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.vc.data_integrity
2025-04-09 11:11:54,990 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.wallet
2025-04-09 11:11:55,030 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.wallet.keys
2025-04-09 11:11:55,032 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.anoncreds
2025-04-09 11:11:55,033 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.anoncreds.default.did_indy
2025-04-09 11:11:55,033 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.anoncreds.default.did_web
2025-04-09 11:11:55,033 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.anoncreds.default.legacy_indy
2025-04-09 11:11:55,034 acapy_agent.core.plugin_registry DEBUG Loaded module: acapy_agent.revocation_anoncreds
2025-04-09 11:11:55,201 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::record::connections::deleted'), processor <bound method PeerDID3Resolver.remove_record_for_deleted_conn of <acapy_agent.resolver.default.peer3.PeerDID3Resolver object at 0x7fa6369c12e0>>
Successfully registered DIDIndyRegistry
Successfully registered DIDWebRegistry
Successfully registered LegacyIndyRegistry
2025-04-09 11:11:55,298 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::published.*'), processor <function on_revocation_published at 0x7fa6367f6480>
2025-04-09 11:11:55,299 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::clear-pending.*'), processor <function on_pending_cleared at 0x7fa6367f6520>
2025-04-09 11:11:55,303 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::published.*'), processor <function on_revocation_published at 0x7fa6367f6980>
2025-04-09 11:11:55,303 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::clear-pending.*'), processor <function on_pending_cleared at 0x7fa6367f7d80>
2025-04-09 11:11:55,365 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::core::startup?$'), processor <function on_startup_event at 0x7fa63656e5c0>
2025-04-09 11:11:55,365 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::core::shutdown?$'), processor <function on_shutdown_event at 0x7fa63659aca0>
2025-04-09 11:11:55,391 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::ENDORSE_DID::(.*)?$'), processor <function on_register_nym_event at 0x7fa636e9b6a0>
2025-04-09 11:11:55,391 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::credential-definition::finished'), processor <bound method DefaultRevocationSetup.on_cred_def of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7fa6367072f0>>
2025-04-09 11:11:55,391 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::revocation-registry-definition::finished'), processor <bound method DefaultRevocationSetup.on_rev_reg_def of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7fa6367072f0>>
2025-04-09 11:11:55,391 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::revocation-list::finished'), processor <bound method DefaultRevocationSetup.on_rev_list of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7fa6367072f0>>
2025-04-09 11:11:55,391 acapy_agent.config.ledger INFO Fetching genesis transactions from: http://ledger-browser:8000/genesis
2025-04-09 11:11:55,406 acapy_agent.core.profile INFO Create profile manager: askar-anoncreds
2025-04-09 11:11:57,466 acapy_agent.ledger.indy_vdr DEBUG Opening the pool ledger
2025-04-09 11:11:57,476 acapy_agent.ledger.indy_vdr DEBUG Wrote pool ledger config 'default'
2025-04-09 11:11:57,712 acapy_agent.multitenant.manager_provider INFO Create multitenant manager: single-wallet-askar
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::webhook::(.*)$'), processor <bound method AdminServer._on_webhook_event of <acapy_agent.admin.server.AdminServer object at 0x7fa63649cd40>>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::record::([^:]*)(?:::.*)?$'), processor <bound method AdminServer._on_record_event of <acapy_agent.admin.server.AdminServer object at 0x7fa63649cd40>>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::basicmessage::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f060>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::problem_report'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f100>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::ping::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f1a0>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::ping::response_received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f240>
2025-04-09 11:11:58,283 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f2e0>
2025-04-09 11:11:58,284 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::get\\-active\\-menu'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f380>
2025-04-09 11:11:58,284 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::perform\\-menu\\-action'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f420>
2025-04-09 11:11:58,284 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::keylist::updated'), processor <function AdminServer.start.<locals>.<lambda> at 0x7fa63491f4c0>

::::::::::::::::::::::::::::::::::::::::::::::
::               Multitenant                ::
::                                          ::
::                                          ::
:: Inbound Transports:                      ::
::                                          ::
::   - http://0.0.0.0:3020/                  ::
::                                          ::
:: Outbound Transports:                     ::
::                                          ::
::   - http                                 ::
::   - https                                ::
::                                          ::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
:: Administration API:                      ::
::                                          ::
::   - http://0.0.0.0:3021/                  ::
::                                          ::
::                               ver: 1.2.4 ::
::::::::::::::::::::::::::::::::::::::::::::::

::                             ⚠ DEPRECATION NOTICE: ⚠                              ::
:: -------------------------------------------------------------------------------- ::
Listening...

:: Receiving a core DIDComm protocol with the `did:sov:BzCbsNYhMrjHiqZDTUASHg;spec` ::
:: prefix is deprecated. All parties sending this prefix should be notified that    ::
:: support for receiving such messages will be removed in a future release. Use     ::
:: https://didcomm.org/ instead.                                                    ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0160: Connection Protocol is deprecated and support will be removed in ::
:: a future release; use RFC 0023: DID Exchange instead.                            ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0036: Issue Credential 1.0 is deprecated and support will be removed   ::
:: in a future release; use RFC 0453: Issue Credential 2.0 instead.                 ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0037: Present Proof 1.0 is deprecated and support will be removed in a ::
:: future release; use RFC 0454: Present Proof 2.0 instead.                         ::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

2025-04-09 11:11:58,321 acapy_agent.core.conductor INFO Existing acapy_version storage record found, version set to v1.2.4
2025-04-09 11:11:58,402 acapy_agent.core.event_bus DEBUG Notifying subscribers: <Event topic=acapy::core::startup, payload={}>
2025-04-09 11:11:58,417 acapy_agent.utils.endorsement_setup INFO No endorser DID, can connect, but can't setup connection metadata.
2025-04-09 11:12:02,718 acapy_agent.ledger.indy_vdr DEBUG Closing pool ledger after timeout
2025-04-09 11:12:04,971 acapy_agent.admin.server DEBUG Incoming request: GET /status/live
2025-04-09 11:12:04,971 acapy_agent.admin.server DEBUG Match info: <MatchInfo {}: <ResourceRoute [GET] <PlainResource  /status/live> -> <function liveliness_handler at 0x7fa6375aa3e0>>
2025-04-09 11:12:04,971 acapy_agent.admin.server DEBUG Body: None
2025-04-09 11:12:05,001 aiohttp.access INFO 127.0.0.6 [09/Apr/2025:11:12:04 +0000] "GET /status/live HTTP/1.1" 200 194 "-" "kube-probe/1.32"
2025-04-09 11:12:05,008 acapy_agent.admin.server DEBUG Incoming request: GET /status/ready
2025-04-09 11:12:05,009 acapy_agent.admin.server DEBUG Match info: <MatchInfo {}: <ResourceRoute [GET] <PlainResource  /status/ready> -> <function readiness_handler at 0x7fa6375aa520>>
2025-04-09 11:12:05,009 acapy_agent.admin.server DEBUG Body: None
```
___
^ Notes:
- fairly sparse logging (first log = loading a module; nothing about start up)
- includes prints
- server logs debug info for /status/ requests

___

Startup logs after (debug level):
```
2025-04-09 11:44:45,178 acapy_agent.core.conductor DEBUG Starting setup of the Conductor
2025-04-09 11:44:45,178 acapy_agent.config.default_context DEBUG Building new injection context
2025-04-09 11:44:45,179 acapy_agent.config.default_context DEBUG Begin binding providers to context
2025-04-09 11:44:45,179 acapy_agent.config.default_context DEBUG Initializing plugin registry
2025-04-09 11:44:45,179 acapy_agent.core.plugin_registry DEBUG Registering package: acapy_agent.protocols
2025-04-09 11:44:45,179 acapy_agent.utils.classloader DEBUG Scanning subpackages under package acapy_agent.protocols
2025-04-09 11:44:45,179 acapy_agent.utils.classloader DEBUG Extracted main package: acapy_agent, sub-package: protocols
2025-04-09 11:44:45,180 acapy_agent.utils.classloader DEBUG 17 sub-packages found under acapy_agent: ['acapy_agent.protocols.did_rotate', 'acapy_agent.protocols.present_proof', 'acapy_agent.protocols.routing', 'acapy_agent.protocols.tests', 'acapy_agent.protocols.didexchange', 'acapy_agent.protocols.introduction', 'acapy_agent.protocols.coordinate_mediation', 'acapy_agent.protocols.revocation_notification', 'acapy_agent.protocols.problem_report', 'acapy_agent.protocols.notification', 'acapy_agent.protocols.trustping', 'acapy_agent.protocols.issue_credential', 'acapy_agent.protocols.actionmenu', 'acapy_agent.protocols.out_of_band', 'acapy_agent.protocols.endorse_transaction', 'acapy_agent.protocols.basicmessage', 'acapy_agent.protocols.discovery']
2025-04-09 11:44:45,180 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.did_rotate
2025-04-09 11:44:45,180 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.present_proof
2025-04-09 11:44:45,181 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.routing
2025-04-09 11:44:45,181 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.didexchange
2025-04-09 11:44:45,181 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.introduction
2025-04-09 11:44:45,181 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.coordinate_mediation
2025-04-09 11:44:45,182 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.revocation_notification
2025-04-09 11:44:45,182 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.problem_report
2025-04-09 11:44:45,182 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.notification
2025-04-09 11:44:45,183 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.trustping
2025-04-09 11:44:45,183 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.issue_credential
2025-04-09 11:44:45,183 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.actionmenu
2025-04-09 11:44:45,183 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.out_of_band
2025-04-09 11:44:45,184 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.endorse_transaction
2025-04-09 11:44:45,184 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.basicmessage
2025-04-09 11:44:45,185 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.protocols.discovery
2025-04-09 11:44:45,185 acapy_agent.config.default_context INFO Registering default plugins
2025-04-09 11:44:45,208 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.holder
2025-04-09 11:44:45,237 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.ledger
2025-04-09 11:44:45,245 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.connections
2025-04-09 11:44:45,250 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.messaging.jsonld
2025-04-09 11:44:45,250 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.resolver
2025-04-09 11:44:45,258 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.settings
2025-04-09 11:44:45,258 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.vc
2025-04-09 11:44:45,348 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.vc.data_integrity
2025-04-09 11:44:45,400 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.wallet
2025-04-09 11:44:45,403 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.wallet.keys
2025-04-09 11:44:45,407 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.did.indy
2025-04-09 11:44:45,407 acapy_agent.config.default_context INFO Registering askar plugins
2025-04-09 11:44:45,414 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.messaging.credential_definitions
2025-04-09 11:44:45,420 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.messaging.schemas
2025-04-09 11:44:45,436 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.revocation
2025-04-09 11:44:45,437 acapy_agent.config.default_context INFO Registering anoncreds plugins
2025-04-09 11:44:45,437 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.anoncreds
2025-04-09 11:44:45,437 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.anoncreds.default.did_indy
2025-04-09 11:44:45,438 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.anoncreds.default.did_web
2025-04-09 11:44:45,438 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.anoncreds.default.legacy_indy
2025-04-09 11:44:45,460 acapy_agent.core.plugin_registry DEBUG Registered plugin: acapy_agent.revocation_anoncreds
2025-04-09 11:44:45,460 acapy_agent.core.plugin_registry DEBUG Initializing plugin context for 35 plugins
2025-04-09 11:44:45,462 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.present_proof.v1_0.controller
2025-04-09 11:44:45,462 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.present_proof.v1_0.controller
2025-04-09 11:44:45,462 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.present_proof.v2_0.controller
2025-04-09 11:44:45,462 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.present_proof.v2_0.controller
2025-04-09 11:44:45,463 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.coordinate_mediation.v1_0.controller
2025-04-09 11:44:45,463 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.coordinate_mediation.v1_0.controller
2025-04-09 11:44:45,464 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.issue_credential.v1_0.controller
2025-04-09 11:44:45,464 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.issue_credential.v1_0.controller
2025-04-09 11:44:45,464 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.issue_credential.v2_0.controller
2025-04-09 11:44:45,464 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.issue_credential.v2_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.actionmenu.v1_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.actionmenu.v1_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.out_of_band.v1_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.out_of_band.v1_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.endorse_transaction.v1_0.controller
2025-04-09 11:44:45,465 acapy_agent.utils.classloader DEBUG Successfully loaded class Controller from module acapy_agent.protocols.endorse_transaction.v1_0.controller
2025-04-09 11:44:45,468 acapy_agent.utils.classloader DEBUG Successfully loaded class LegacyPeerDIDResolver from module acapy_agent.resolver.default.legacy_peer
2025-04-09 11:44:45,468 acapy_agent.utils.classloader DEBUG Successfully loaded class KeyDIDResolver from module acapy_agent.resolver.default.key
2025-04-09 11:44:45,469 acapy_agent.utils.classloader DEBUG Successfully loaded class JwkDIDResolver from module acapy_agent.resolver.default.jwk
2025-04-09 11:44:45,469 acapy_agent.utils.classloader DEBUG Successfully loaded class IndyDIDResolver from module acapy_agent.resolver.default.indy
2025-04-09 11:44:45,470 acapy_agent.utils.classloader DEBUG Successfully loaded class WebDIDResolver from module acapy_agent.resolver.default.web
2025-04-09 11:44:45,520 acapy_agent.utils.classloader DEBUG Successfully loaded class WebvhDIDResolver from module acapy_agent.resolver.default.webvh
2025-04-09 11:44:45,521 acapy_agent.utils.classloader DEBUG Successfully loaded class PeerDID1Resolver from module acapy_agent.resolver.default.peer1
2025-04-09 11:44:45,522 acapy_agent.utils.classloader DEBUG Successfully loaded class PeerDID2Resolver from module acapy_agent.resolver.default.peer2
2025-04-09 11:44:45,522 acapy_agent.utils.classloader DEBUG Successfully loaded class PeerDID3Resolver from module acapy_agent.resolver.default.peer3
2025-04-09 11:44:45,522 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::record::connections::deleted'), processor <bound method PeerDID3Resolver.remove_record_for_deleted_conn of <acapy_agent.resolver.default.peer3.PeerDID3Resolver object at 0x7f184bac7140>>
2025-04-09 11:44:45,522 acapy_agent.utils.classloader DEBUG Successfully loaded class PeerDID4Resolver from module acapy_agent.resolver.default.peer4
2025-04-09 11:44:45,528 acapy_agent.utils.classloader DEBUG Successfully loaded class DIDIndyRegistry from module acapy_agent.anoncreds.default.did_indy.registry
2025-04-09 11:44:45,528 acapy_agent.anoncreds.default.did_indy.registry INFO Successfully registered DIDIndyRegistry
2025-04-09 11:44:45,529 acapy_agent.utils.classloader DEBUG Successfully loaded class DIDWebRegistry from module acapy_agent.anoncreds.default.did_web.registry
2025-04-09 11:44:45,529 acapy_agent.anoncreds.default.did_web.registry INFO Successfully registered DIDWebRegistry
2025-04-09 11:44:45,529 acapy_agent.utils.classloader DEBUG Successfully loaded class LegacyIndyRegistry from module acapy_agent.anoncreds.default.legacy_indy.registry
2025-04-09 11:44:45,530 acapy_agent.anoncreds.default.legacy_indy.registry INFO Successfully registered LegacyIndyRegistry
2025-04-09 11:44:45,531 acapy_agent.core.plugin_registry DEBUG Registering protocol events for 35 plugins
2025-04-09 11:44:45,597 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::published.*'), processor <function on_revocation_published at 0x7f184b4acc20>
2025-04-09 11:44:45,597 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::clear-pending.*'), processor <function on_pending_cleared at 0x7f184b4accc0>
2025-04-09 11:44:45,600 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::published.*'), processor <function on_revocation_published at 0x7f184b4ad120>
2025-04-09 11:44:45,601 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::clear-pending.*'), processor <function on_pending_cleared at 0x7f184b4ae520>
2025-04-09 11:44:45,664 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::core::startup?$'), processor <function on_startup_event at 0x7f184b41d440>
2025-04-09 11:44:45,664 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::core::shutdown?$'), processor <function on_shutdown_event at 0x7f184b249bc0>
2025-04-09 11:44:45,697 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::ENDORSE_DID::(.*)?$'), processor <function on_register_nym_event at 0x7f184bd8f1a0>
2025-04-09 11:44:45,697 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::CRED_DEF::(.*)?$'), processor <function on_cred_def_event at 0x7f184bb8b880>
2025-04-09 11:44:45,697 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::SCHEMA::(.*)?$'), processor <function on_schema_event at 0x7f184bba60c0>
2025-04-09 11:44:45,698 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::REGISTRY_INIT.*'), processor <function on_revocation_registry_init_event at 0x7f184bbea160>
2025-04-09 11:44:45,698 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::REGISTRY_ENDORSED.*'), processor <function on_revocation_registry_endorsed_event at 0x7f184ba10a40>
2025-04-09 11:44:45,698 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::SEND_ENTRY.*'), processor <function on_revocation_entry_event at 0x7f184ba10680>
2025-04-09 11:44:45,698 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::REVOCATION::REV_REG_ENTRY_TXN_FAILED.*'), processor <function on_rev_reg_entry_txn_failed at 0x7f184ba10c20>
2025-04-09 11:44:45,699 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::credential-definition::finished'), processor <bound method DefaultRevocationSetup.on_cred_def of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7f184b423620>>
2025-04-09 11:44:45,699 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::revocation-registry-definition::finished'), processor <bound method DefaultRevocationSetup.on_rev_reg_def of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7f184b423620>>
2025-04-09 11:44:45,699 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('anoncreds::revocation-list::finished'), processor <bound method DefaultRevocationSetup.on_rev_list of <acapy_agent.anoncreds.revocation_setup.DefaultRevocationSetup object at 0x7f184b423620>>
2025-04-09 11:44:45,699 acapy_agent.core.conductor DEBUG Context built successfully
2025-04-09 11:44:45,699 acapy_agent.core.conductor DEBUG Genesis transactions/configurations found. Fetching genesis transactions
2025-04-09 11:44:45,699 acapy_agent.config.ledger DEBUG Getting genesis transactions from settings
2025-04-09 11:44:45,699 acapy_agent.config.ledger DEBUG Genesis transactions from settings: absent
2025-04-09 11:44:45,699 acapy_agent.config.ledger DEBUG No genesis transactions found in settings
2025-04-09 11:44:45,700 acapy_agent.config.ledger INFO Fetching genesis transactions from: http://ledger-browser:8000/genesis
2025-04-09 11:44:45,715 acapy_agent.config.ledger DEBUG Storing genesis transactions in settings
2025-04-09 11:44:45,715 acapy_agent.core.conductor DEBUG Configuring the root profile and setting up public DID
2025-04-09 11:44:45,716 acapy_agent.core.profile INFO Create profile manager: askar-anoncreds
2025-04-09 11:44:45,716 acapy_agent.utils.classloader DEBUG Successfully loaded class AskarAnonProfileManager from module acapy_agent.askar.profile_anon
2025-04-09 11:44:47,264 acapy_agent.config.wallet INFO Opened existing profile - Profile name: multitenant, backend: askar-anoncreds
2025-04-09 11:44:47,272 acapy_agent.utils.classloader DEBUG Successfully loaded class AskarWallet from module acapy_agent.wallet.askar
2025-04-09 11:44:47,281 acapy_agent.core.conductor DEBUG Root profile configured successfully
2025-04-09 11:44:47,281 acapy_agent.config.ledger DEBUG Configuring ledger for profile multitenant and public_did None
2025-04-09 11:44:47,287 acapy_agent.ledger.indy_vdr DEBUG Opening the pool ledger
2025-04-09 11:44:47,293 acapy_agent.ledger.indy_vdr DEBUG Wrote pool ledger config 'default'
2025-04-09 11:44:47,350 acapy_agent.config.ledger DEBUG Checking transaction author agreement
2025-04-09 11:44:47,389 acapy_agent.config.ledger INFO Ledger configuration complete
2025-04-09 11:44:47,389 acapy_agent.core.conductor DEBUG Ledger configured successfully.
2025-04-09 11:44:47,390 acapy_agent.core.conductor DEBUG Transport not disabled. Setting up inbound transports.
2025-04-09 11:44:47,391 acapy_agent.core.conductor DEBUG Inbound transports registered successfully.
2025-04-09 11:44:47,391 acapy_agent.core.conductor DEBUG Setting up outbound transports.
2025-04-09 11:44:47,392 acapy_agent.core.conductor DEBUG Outbound transports registered successfully.
2025-04-09 11:44:47,392 acapy_agent.core.conductor DEBUG Initializing dispatcher.
2025-04-09 11:44:47,392 acapy_agent.core.conductor DEBUG Dispatcher initialized successfully.
2025-04-09 11:44:47,409 acapy_agent.utils.classloader DEBUG Successfully loaded class PackWireFormat from module acapy_agent.transport.pack_format
2025-04-09 11:44:47,409 acapy_agent.core.conductor DEBUG Multitenant is enabled. Binding MultitenantManagerProvider.
2025-04-09 11:44:47,409 acapy_agent.core.conductor DEBUG Binding RouteManagerProvider.
2025-04-09 11:44:47,409 acapy_agent.core.conductor DEBUG Binding OobMessageProcessor.
2025-04-09 11:44:47,409 acapy_agent.core.conductor DEBUG Binding default DocumentLoader.
2025-04-09 11:44:47,411 acapy_agent.core.conductor DEBUG Admin API is enabled. Attempting to register admin server.
2025-04-09 11:44:47,411 acapy_agent.multitenant.manager_provider INFO Create multitenant manager: single-wallet-askar
2025-04-09 11:44:47,411 acapy_agent.utils.classloader DEBUG Successfully loaded class SingleWalletAskarMultitenantManager from module acapy_agent.multitenant.single_wallet_askar_manager
2025-04-09 11:44:47,411 acapy_agent.core.conductor DEBUG Admin server registered on 0.0.0.0:3021
2025-04-09 11:44:47,412 acapy_agent.core.conductor DEBUG Starting the Conductor agent.
[wait-for-ledger-browser] 
2025-04-09 11:44:47,415 acapy_agent.utils.classloader DEBUG Successfully loaded class AskarStorage from module acapy_agent.storage.askar
2025-04-09 11:44:47,421 acapy_agent.core.conductor DEBUG Wallet type validated.
2025-04-09 11:44:47,422 acapy_agent.core.conductor DEBUG Transport not disabled. Starting inbound transports.
2025-04-09 11:44:47,422 acapy_agent.core.conductor DEBUG Inbound transports started successfully.
2025-04-09 11:44:47,422 acapy_agent.core.conductor DEBUG Starting outbound transports.
2025-04-09 11:44:47,422 acapy_agent.core.conductor DEBUG Outbound transports started successfully.
2025-04-09 11:44:47,422 acapy_agent.core.conductor DEBUG Admin server present. Starting admin server.
2025-04-09 11:44:47,422 acapy_agent.core.plugin_registry DEBUG Registering admin routes for 35 plugins
2025-04-09 11:44:47,945 acapy_agent.core.plugin_registry DEBUG Post-processing routes for 35 plugins
2025-04-09 11:44:47,946 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::webhook::(.*)$'), processor <bound method AdminServer._on_webhook_event of <acapy_agent.admin.server.AdminServer object at 0x7f184b2effe0>>
2025-04-09 11:44:47,946 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('^acapy::record::([^:]*)(?:::.*)?$'), processor <bound method AdminServer._on_record_event of <acapy_agent.admin.server.AdminServer object at 0x7f184b2effe0>>
2025-04-09 11:44:47,946 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::basicmessage::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd5080>
2025-04-09 11:44:47,946 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::problem_report'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd4fe0>
2025-04-09 11:44:47,946 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::ping::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd5120>
2025-04-09 11:44:47,947 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::ping::response_received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd51c0>
2025-04-09 11:44:47,947 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::received'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd5260>
2025-04-09 11:44:47,947 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::get\\-active\\-menu'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd5300>
2025-04-09 11:44:47,947 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::actionmenu::perform\\-menu\\-action'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd53a0>
[nc-postgres] 
2025-04-09 11:44:47,947 acapy_agent.core.event_bus DEBUG Subscribed: topic re.compile('acapy::keylist::updated'), processor <function AdminServer.start.<locals>.<lambda> at 0x7f1849bd5440>

::::::::::::::::::::::::::::::::::::::::::::::
::               Multitenant                ::
::                                          ::
::                                          ::
:: Inbound Transports:                      ::
::                                          ::
2025-04-09 11:44:47,955 acapy_agent.core.conductor DEBUG Admin server started successfully.
2025-04-09 11:44:47,955 acapy_agent.core.conductor DEBUG Admin responder bound to injector.
2025-04-09 11:44:47,955 acapy_agent.core.conductor DEBUG Agent label: Multitenant
::   - http://0.0.0.0:3020/                  ::
::                                          ::
:: Outbound Transports:                     ::
::                                          ::
::   - http                                 ::
::   - https                                ::
::                                          ::
:: Administration API:                      ::
::                                          ::
::   - http://0.0.0.0:3021/                  ::
::                                          ::
::                            ver: 1.3.0rc1 ::
::::::::::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

Listening...

::                             ⚠ DEPRECATION NOTICE: ⚠                              ::
:: -------------------------------------------------------------------------------- ::
:: Receiving a core DIDComm protocol with the `did:sov:BzCbsNYhMrjHiqZDTUASHg;spec` ::
:: prefix is deprecated. All parties sending this prefix should be notified that    ::
:: support for receiving such messages will be removed in a future release. Use     ::
:: https://didcomm.org/ instead.                                                    ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0160: Connection Protocol is deprecated and support will be removed in ::
:: a future release; use RFC 0023: DID Exchange instead.                            ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0036: Issue Credential 1.0 is deprecated and support will be removed   ::
:: in a future release; use RFC 0453: Issue Credential 2.0 instead.                 ::
:: -------------------------------------------------------------------------------- ::
:: Aries RFC 0037: Present Proof 1.0 is deprecated and support will be removed in a ::
:: future release; use RFC 0454: Present Proof 2.0 instead.                         ::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

2025-04-09 11:44:47,956 acapy_agent.core.conductor DEBUG Recording ACA-Py version in wallet if needed.
2025-04-09 11:44:47,966 acapy_agent.core.conductor INFO Existing acapy_version storage record found, version set to v1.3.0rc1
2025-04-09 11:44:47,966 acapy_agent.core.conductor DEBUG Force upgrade flag: False, From version config: None
2025-04-09 11:44:47,966 acapy_agent.core.conductor DEBUG Determined from_version: v1.3.0rc1
2025-04-09 11:44:47,970 acapy_agent.core.conductor DEBUG Available upgrade versions: []
2025-04-09 11:44:47,971 acapy_agent.core.conductor DEBUG Mediation invite provided: None
2025-04-09 11:44:47,981 acapy_agent.core.conductor DEBUG Mediation invite record retrieved successfully.
2025-04-09 11:44:47,982 acapy_agent.core.conductor DEBUG Checking for wallet upgrades in progress.
2025-04-09 11:44:47,982 acapy_agent.utils.classloader DEBUG Successfully loaded class AskarStorageSearch from module acapy_agent.storage.askar
2025-04-09 11:44:48,007 acapy_agent.core.conductor DEBUG Wallet upgrades check completed.
2025-04-09 11:44:48,007 acapy_agent.core.conductor DEBUG Notifying protocols of startup status.
2025-04-09 11:44:48,007 acapy_agent.core.event_bus DEBUG Notifying subscribers: <Event topic=acapy::core::startup, payload={}>
2025-04-09 11:44:48,023 acapy_agent.utils.endorsement_setup INFO No endorser DID, can connect, but can't setup connection metadata.
2025-04-09 11:44:48,024 acapy_agent.core.conductor DEBUG Startup notification sent.
2025-04-09 11:44:52,413 acapy_agent.ledger.indy_vdr DEBUG Closing pool ledger after timeout
```

^ Notes:
- much more verbose (which can be a good thing for debug level) - includes starting up info and config inputs
- No more print() calls (apart from banner)
- Server no longer logs match info for /status/ checks:
```
2025-04-09 11:12:05,008 acapy_agent.admin.server DEBUG Incoming request: GET /status/ready
2025-04-09 11:12:05,009 acapy_agent.admin.server DEBUG Match info: <MatchInfo {}: <ResourceRoute [GET] <PlainResource  /status/ready> -> <function readiness_handler at 0x7fa6375aa520>>
2025-04-09 11:12:05,009 acapy_agent.admin.server DEBUG Body: None
```

ff137 commented:
Just included a commit so that status checks will still be logged as an incoming request when in debug mode (might be confusing if status checks aren't logged at all) - they're just logged without the redundant match info, or `Body: None`

ff137 commented:
I think the print statements in the banner can be replaced with an info log - where instead of each line being printed, we can just build up a list of strings to be printed - join them with new lines, and info log it. (That'll avoid other log lines interrupting the banner, as it currently does). That could be an effective alternative to suppressing banner logs, as was proposed here: https://github.com/openwallet-foundation/acapy/issues/3202

swcurran commented:
Good idea about the banner, but can be for another. Let’s get this in.
